backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/FlatrateDAO.java

```diff
@@ -594,19 +594,22 @@ public List<I_C_Invoice_Clearing_Alloc> retrieveOpenClearingAllocs(final I_C_Fla
 	@Override
 	public final List<I_C_Flatrate_DataEntry> retrieveInvoicingEntries(
 			final I_C_Flatrate_Term flatrateTerm,
-			final Timestamp dateFrom, final Timestamp dateTo,
+			final @NonNull LocalDateAndOrgId dateFrom, @NonNull final LocalDateAndOrgId dateTo,
 			final UomId uomId)
 	{
 		final List<I_C_Flatrate_DataEntry> result = new ArrayList<>();
 
 		final IFlatrateDAO flatrateDB = Services.get(IFlatrateDAO.class);
 		final List<I_C_Flatrate_DataEntry> entriesToCorrect = flatrateDB.retrieveDataEntries(flatrateTerm, X_C_Flatrate_DataEntry.TYPE_Invoicing_PeriodBased, uomId);
 
+		final Timestamp dateFromTimestamp = dateFrom.toTimestamp(orgDAO::getTimeZone);
+		final Timestamp dateToTimestamp = dateTo.toTimestamp(orgDAO::getTimeZone);
+
 		for (final I_C_Flatrate_DataEntry entryToCorrect : entriesToCorrect)
 		{
 			final I_C_Period entryPeriod = entryToCorrect.getC_Period();
-			if (entryPeriod.getEndDate().before(dateFrom) // entryPeriod ends before dateFrom
-					|| entryPeriod.getStartDate().after(dateTo))      // entryPeriod begins after dateTo
+			if (entryPeriod.getEndDate().before(dateFromTimestamp) // entryPeriod ends before dateFrom
```

Comment:
I would rather implement a compareTo method in LocalDateOrgId and use it instead of comparing timestamps

Reply:
Although I dismissed the initial solution of using `LocalDateOrgId#compareTo`, re-reading how we construct the LocalDateOrgId, this makes the most sense.